src/js/media-controller.js

```diff
@@ -532,16 +488,73 @@ const propagateMediaState = (els, stateName, val) => {
   });
 };
 
-const monitorMediaUIElementDescendantsOf = (root, associateCallback, unassociateCallback) => {
+/**
+ * 
+ * @description This function will monitor the rootNode for any Media State Receiver descendants
+ * that are already present, added, or removed, invoking the relevant callback function for each
+ * case.
+ * 
+ * @param {HTMLElement} rootNode 
+ * @param {function} registerMediaStateReceiver
+ * @param {function} unregisterMediaStateReceiver
+ * @returns An unsubscribe method, used to stop monitoring descendants of rootNode and to unregister its descendants
+ * 
+ */
+const monitorForMediaStateReceivers = (rootNode, registerMediaStateReceiver, unregisterMediaStateReceiver) => {
+
+  // First traverse the tree to register any current Media State Receivers
+  traverseForMediaStateReceivers(rootNode, registerMediaStateReceiver);
+  
+  // Monitor for any event-based requests from descendants to register/unregister as a Media State Receiver
+  const registerMediaStateReceiverHandler = (evt) => {
+    const el = evt?.composedPath()[0] ?? evt.target;
+    registerMediaStateReceiver(el);
+  };
+
+  const unregisterMediaStateReceiverHandler = (evt) => {
+    const el = evt?.composedPath()[0] ?? evt.target;
+    unregisterMediaStateReceiver(el);
+  };
+
+  rootNode.addEventListener(MediaUIEvents.REGISTER_MEDIA_STATE_RECEIVER, registerMediaStateReceiverHandler);
+  rootNode.addEventListener(MediaUIEvents.UNREGISTER_MEDIA_STATE_RECEIVER, unregisterMediaStateReceiverHandler);
+
+  // Observe any changes to the DOM for any descendants that are identifiable as Media State Receivers
+  // and register or unregister them, depending on the change that occurred.
   const mutationCallback = (mutationsList, _observer) => {
-    const { addedNodes, removedNodes } = toNextMediaUIElementsState(mutationsList);
-    addedNodes.forEach(associateCallback);
-    removedNodes.forEach(unassociateCallback);
+    mutationsList.forEach(mutationRecord => {
+      const { addedNodes = [], removedNodes = [], type, target, attributeName } = mutationRecord;
+      if (type === 'childList') {
+        // For each added node, register any Media State Receiver descendants (including itself)
+        Array.prototype.forEach.call(addedNodes, node => traverseForMediaStateReceivers(node, registerMediaStateReceiver));
+        // For each removed node, unregister any Media State Receiver descendants (including itself)
+        Array.prototype.forEach.call(removedNodes, node => traverseForMediaStateReceivers(node, unregisterMediaStateReceiver));
+      } else if (type === 'attributes' && attributeName === MediaUIAttributes.MEDIA_CHROME_ATTRIBUTES) {
+        if (isMediaStateReceiver(target)) {
+          // Changed from a "non-Media State Receiver" to a Media State Receiver: register it.
```

Comment:
This isn't necessarily true if the attr list just changed, right? Though we're checking if it already is registered later so not an issue.

Reply:
Yup that's exactly right. I'll update the code comment for further clarity.